src/probnum/diffeq/_perturbsolve_ivp.py

```diff
@@ -0,0 +1,191 @@
+              steps are taken using the third-order accurate formula (local
+              extrapolation is done). A cubic Hermite polynomial is used for the
+              dense output. Can be applied in the complex domain.
+
+        Other integrators are not supported currently.
+    perturb
+        Which perturbation style to use.
+        Currently, the following methods are supported:
+
+            * `step-lognormal`: Perturbed-step (aka random time-step numerical integration) method
+              with lognormally distributed perturbation of the step-size [1]_.
+            * `step-uniform`: Perturbed-step (aka random time-step numerical integration) method
+              with lognormally distributed perturbation of the step-size [1]_.
+
+        Other integrators are not supported currently.
+    noise_scale
+        Scale of the perturbation. Optional. Default is 10.0. The magnitude of this parameter
+        significantly impacts the width of the posterior.
+    adaptive :
+        Whether to use adaptive steps or not. Default is `True`.
+    atol :
+        Absolute tolerance  of the adaptive step-size selection scheme.
+        Optional. Default is ``1e-4``.
+    rtol :
+        Relative tolerance   of the adaptive step-size selection scheme.
+        Optional. Default is ``1e-4``.
+    step :
+        Step size. If atol and rtol are not specified, this step-size is used for a fixed-step ODE solver.
+        If they are specified, this only affects the first step. Optional.
+        Default is None, in which case the first step is chosen as prescribed by :meth:`propose_firststep`.
+
+    Examples
+    --------
+    >>> from probnum.diffeq import perturbsolve_ivp
+    >>> import numpy as np
+
+    Solve a simple logistic ODE with fixed steps.
+
+    >>> rng = np.random.default_rng(seed=2)
+    >>>
+    >>> def f(t, x):
+    ...     return 4*x*(1-x)
+    >>>
+    >>> y0 = np.array([0.15])
+    >>> t0, tmax = 0., 1.5
+    >>> solution = perturbsolve_ivp(f, t0, tmax, y0, rng=rng, step=0.25, method="RK23", adaptive=False)
+    >>> print(np.round(solution.states.mean, 3))
+    [[0.15 ]
+     [0.292]
+     [0.497]
+     [0.766]
+     [0.874]
+     [0.955]
+     [0.987]]
+
+    Each solution is the result of a randomly-perturbed call the an underlying Runge-Kutta solver.
+    Therefore, if you call it again, the output will be different:
+
+    >>> other_solution = perturbsolve_ivp(f, t0, tmax, y0, rng=rng, step=0.25, method="RK23", adaptive=False)
+    >>> print(np.round(other_solution.states.mean, 3))
+    [[0.15 ]
+     [0.282]
+     [0.448]
+     [0.657]
+     [0.853]
+     [0.945]
+     [0.975]]
+
+    Other methods, such as `RK45` (as well as other perturbation styles) are easily accessible.
+    Let us solve the same equation, with an adaptive RK45 solver that uses lognormally perturbed steps.
+
+    >>> solution = perturbsolve_ivp(f, t0, tmax, y0, rng=rng, atol=1e-5, rtol=1e-6, method="RK45", adaptive=True)
```

Comment:
`` solution = perturbsolve_ivp(f, t0, tmax, y0, rng=rng, atol=1e-5, rtol=1e-6, method="RK45", perturb="step-lognormal", adaptive=True)``

Reply:
(you forgot to set the perturbation function :))